plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaRecordSet.java

```diff
@@ -173,61 +173,28 @@ private boolean nextRow(ConsumerRecord<byte[], byte[]> message)
-                            break;
-                        case KEY_FIELD:
-                            currentRowValuesMap.put(columnHandle, bytesValueProvider(keyData));
-                            break;
-                        case KEY_LENGTH_FIELD:
-                            currentRowValuesMap.put(columnHandle, longValueProvider(keyData.length));
-                            break;
-                        case OFFSET_TIMESTAMP_FIELD:
-                            timeStamp *= MICROSECONDS_PER_MILLISECOND;
-                            currentRowValuesMap.put(columnHandle, longValueProvider(timeStamp));
-                            break;
-                        case KEY_CORRUPT_FIELD:
-                            currentRowValuesMap.put(columnHandle, booleanValueProvider(decodedKey.isEmpty()));
-                            break;
-                        case HEADERS_FIELD:
-                            currentRowValuesMap.put(columnHandle, headerMapValueProvider((MapType) columnHandle.getType(), message.headers()));
-                            break;
-                        case MESSAGE_CORRUPT_FIELD:
-                            currentRowValuesMap.put(columnHandle, booleanValueProvider(decodedValue.isEmpty()));
-                            break;
-                        case PARTITION_ID_FIELD:
-                            currentRowValuesMap.put(columnHandle, longValueProvider(message.partition()));
-                            break;
-                        default:
-                            throw new IllegalArgumentException("unknown internal field " + columnHandle.getName());
-                    }
-                }
-            }
+            Map<ColumnHandle, FieldValueProvider> currentRowValuesMap = columnHandles.stream()
+                    .filter(KafkaColumnHandle::isInternal)
+                    .collect(toMap(identity(), columnHandle -> switch (columnHandle.getName()) {
+                        case PARTITION_OFFSET_FIELD -> longValueProvider(message.offset());
+                        case MESSAGE_FIELD -> bytesValueProvider(messageData);
+                        case MESSAGE_LENGTH_FIELD -> longValueProvider(messageData.length);
+                        case KEY_FIELD -> bytesValueProvider(keyData);
+                        case KEY_LENGTH_FIELD -> longValueProvider(keyData.length);
+                        case OFFSET_TIMESTAMP_FIELD -> longValueProvider(timeStamp);
+                        case KEY_CORRUPT_FIELD -> booleanValueProvider(decodedKey.isEmpty());
+                        case HEADERS_FIELD -> headerMapValueProvider((MapType) columnHandle.getType(), message.headers());
+                        case MESSAGE_CORRUPT_FIELD -> booleanValueProvider(decodedValue.isEmpty());
+                        case PARTITION_ID_FIELD -> longValueProvider(message.partition());
+                        default -> throw new IllegalArgumentException("unknown internal field " + columnHandle.getName());
```

Comment:
with switch expression if we leave out default and are missing some enum it'll be a compile time error. Why not use that?